app/containers/RoomItem/index.tsx

```diff
@@ -23,149 +26,106 @@ const attrs = [
 	'displayMode'
 ];
 
-class RoomItemContainer extends React.Component<IRoomItemContainerProps, any> {
-	private roomSubscription: ISubscription | undefined;
-
-	static defaultProps: Partial<IRoomItemContainerProps> = {
-		status: 'offline',
-		getUserPresence: () => {},
-		getRoomTitle: () => 'title',
-		getRoomAvatar: () => '',
-		getIsGroupChat: () => false,
-		getIsRead: () => false,
-		swipeEnabled: true
-	};
-
-	constructor(props: IRoomItemContainerProps) {
-		super(props);
-		this.init();
-	}
-
-	componentDidMount() {
-		const { connected, getUserPresence, id } = this.props;
-		if (connected && this.isDirect) {
-			getUserPresence(id);
-		}
-	}
-
-	shouldComponentUpdate(nextProps: IRoomItemContainerProps) {
-		const { props } = this;
-		return !attrs.every(key => props[key] === nextProps[key]);
-	}
-
-	componentDidUpdate(prevProps: IRoomItemContainerProps) {
-		const { connected, getUserPresence, id } = this.props;
-		if (prevProps.connected !== connected && connected && this.isDirect) {
-			getUserPresence(id);
-		}
-	}
-
-	componentWillUnmount() {
-		if (this.roomSubscription?.unsubscribe) {
-			this.roomSubscription.unsubscribe();
-		}
-	}
-
-	get isGroupChat() {
-		const { item, getIsGroupChat } = this.props;
-		return getIsGroupChat(item);
-	}
-
-	get isDirect() {
-		const {
-			item: { t },
-			id
-		} = this.props;
-		return t === 'd' && id && !this.isGroupChat;
-	}
-
-	init = () => {
-		const { item } = this.props;
-		if (item?.observe) {
-			const observable = item.observe();
-			this.roomSubscription = observable?.subscribe?.(() => {
-				this.forceUpdate();
-			});
-		}
-	};
-
-	onPress = () => {
-		const { item, onPress } = this.props;
-		return onPress(item);
-	};
-
-	onLongPress = () => {
-		const { item, onLongPress } = this.props;
-		if (onLongPress) {
-			return onLongPress(item);
-		}
-	};
-
-	render() {
-		const {
-			item,
-			getRoomTitle,
-			getRoomAvatar,
-			getIsRead,
-			width,
-			toggleFav,
-			toggleRead,
-			hideChannel,
-			theme,
-			isFocused,
-			status,
-			showLastMessage,
-			username,
-			useRealName,
-			swipeEnabled,
-			autoJoin,
-			showAvatar,
-			displayMode
-		} = this.props;
+let roomSubscription: Subscription | null;
+let accessibilityLabel = '';
```

Comment:
Similar to `roomSubscription`, but there's actually no benefit from having it out of the component.